### PR TITLE
[Cache] Add chapter about invalidation, tags, etc.

### PR DESCRIPTION
**Pull request by nicolas-grekas**
This is a subpart of #6858 that tells only about merged features. Please merge asap so we can rebase #6858 on top and focus on its specific bits afterwards :)

### CODE REVIEW

File: components/cache/cache_invalidation.rst [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+Cache Invalidation
+==================
+
+Cache invalidation is the process of removing all cached items related to a
+change in the state of your model. The most basic kind of invalidation is direct
+items deletion. But when the state of a primary resource has spread accross
+several cached items, keeping them in sync can be difficult.
+
+The Symfony Cache component provides three mechanisms to help solve this problem:
```
Comment (javiereguiluz) [inline components/cache/cache_invalidation.rst]: `three mechanisms` -> but the list only contains two :)

File: components/cache/cache_invalidation.rst [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+cache items, as returned by cache adapters::
+
+    $item = $cache->getItem('cache_key');
+    // ...
+    // add one or more tags
+    $item->tag('tag_1');
+    $item->tag(array('tag_2', 'tag_3'));
+    $cache->save($item);
+
+If `$cache` implements :class:`Symfony\\Component\\Cache\\TagAwareAdapterInterface`,
```
Comment (javiereguiluz) [inline components/cache/cache_invalidation.rst]: Double backticks for `$cache`

File: components/cache/cache_invalidation.rst [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+you can invalidate the cached items by calling
+:method:`Symfony\\Component\\Cache\\TagAwareAdapterInterface::invalidateTags`::
+
+    // invalidate all items related to `tag_1`
+    $cache->invalidateTags('tag_2');
```
Comment (javiereguiluz) [inline components/cache/cache_invalidation.rst]: the comment says `tag_1` but the code says `tag_2`

File: components/cache/cache_invalidation.rst [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+    // or invalidate all items related to `tag_1` or `tag_3`
+    $cache->invalidateTags(array('tag_1', 'tag_3'));
+
+    // if you know the cache key, you can of course delete directly
+    $cache->deleteItem('cache_key');``
```
Comment (javiereguiluz) [inline components/cache/cache_invalidation.rst]: Backticks at the end of this line can be removed

File: components/cache/cache_invalidation.rst [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+instantaneous invalidation (time complexity is ``O(N)`` where ``N`` is the number
+of invalidated tags). It needs one or two cache adapters: the first required
+one is used to store cached items; the second optional one is used to store tags
+and their invalidation version number (conceptually similar to their latest
+invalidation date). When only one adapter is used, items and tags are all stored
+in the same place. By using two adapters, you can e.g. store some big cached items
+on the filesystem or in the database and keep tags in a Redis database to sync all
+your fronts and have very fast invalidation checks::
+
+    use Symfony\Component\Cache\Adapter\TagAwareAdapter;
```
Comment (javiereguiluz) [inline components/cache/cache_invalidation.rst]: This is a great example and very well explained!!

**Comment by javiereguiluz** [comment]
👍 

Status: reviewed

File: components/cache/cache_invalidation.rst (hunk 4 above)
Comment (nicolas-grekas) [inline components/cache/cache_invalidation.rst]: comments addressed

**Comment by wouterj** [comment]
Thanks, looks great! I'm sorry we didn't really "merge asap", but at least it is in before the release.

**Pull request merged**